### PR TITLE
fix: move applyPassive to dom package for use in text-field

### PR DESCRIPTION
**Pull request by moog16**
fixes https://github.com/material-components/material-components-web/issues/4460

### CODE REVIEW

File: packages/mdc-ripple/util.ts [hunk 1]
```diff
@@ -86,30 +80,6 @@ export function supportsCssVariables(windowObj: Window, forceRefresh = false): b
   return supportsCssVars;
 }
 
-/**
- * Determine whether the current browser supports passive event listeners, and
- * if so, use them.
- */
-export function applyPassive(globalObj: Window = window, forceRefresh = false):
```
Comment (abhiomkar) [inline packages/mdc-ripple/util.ts]: Seems like only IE does not support `{passive: boolean}` option. Doesn't the browser just ignore this option if not supported. Not sure why we've introduced this check in first place. WDYT?

Comment (moog16) [reply packages/mdc-ripple/util.ts]: I looked into it - https://developers.google.com/web/tools/lighthouse/audits/passive-event-listeners

> However, in browsers that do not support passive event listeners, the third parameter is a boolean to indicate whether the event should bubble or capture. So, using the syntax above may cause unintended consequences.

I think we need to leave this in here if I understand correctly.

File: packages/mdc-dom/events.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * Stores result from applyPassive to avoid redundant processing to detect
+ * passive event listener support.
+ */
+let supportsPassive_: boolean | undefined;
+
+/**
+ * Determine whether the current browser supports passive event listeners, and
+ * if so, use them.
+ */
+export function applyPassive(globalObj: Window = window, forceRefresh = false):
```
Comment (abhiomkar) [inline packages/mdc-dom/events.ts]: Can we also apply this for other component that attach touch events (including, TabScroller, Slider?)

**Comment by moog16** [comment]
depends on https://github.com/material-components/material-components-web/pull/4824

**Pull request merged**